secure-storage/secure-storage-store/src/main/java/com/duckduckgo/securestorage/store/SecureStorageKeyStore.kt

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+
+package com.duckduckgo.securestorage.store
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Base64
+import androidx.core.content.edit
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKey
+import java.lang.Exception
+
+/**
+ * This class provides a way to access and store key related data
+ */
+interface SecureStorageKeyStore {
+    /**
+     * User / Programmatically generated password to be used for L2 encryption
+     */
+    var password: ByteArray?
+
+    /**
+     * Key used for L1 encryption
+     */
+    var l1Key: ByteArray?
+
+    /**
+     * Encrypted key that can be decrypted to be used for L2 encryption
+     */
+    var encryptedL2Key: ByteArray?
+
+    /**
+     * This method can be check if the keystore has support encryption
+     *
+     * @return `true` if all the crypto dependencies needed by description is available and `false` otherwise
+     */
+    fun canUseEncryption(): Boolean
+}
+
+class RealSecureStorageKeyStore constructor(
+    private val context: Context
+) : SecureStorageKeyStore {
+    companion object {
+        const val FILENAME = "com.duckduckgo.securestorage.store"
+        const val KEY_GENERATED_PASSWORD = "KEY_GENERATED_PASSWORD"
+        const val KEY_L1KEY = "KEY_L1KEY"
+        const val KEY_ENCRYPTED_L2KEY = "KEY_ENCRYPTED_L2KEY"
+    }
+
+    private val encryptedPreferences: SharedPreferences? by lazy { encryptedPreferences() }
+
+    @Synchronized
+    private fun encryptedPreferences(): SharedPreferences? {
+        return try {
+            EncryptedSharedPreferences.create(
+                context,
+                FILENAME,
+                MasterKey.Builder(context)
+                    .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+                    .build(),
+                EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+                EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+            )
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    override var password: ByteArray?
+        get() = getValue(KEY_GENERATED_PASSWORD)
+        set(value) {
+            updateValue(KEY_GENERATED_PASSWORD, value)
+        }
+
+    override var l1Key: ByteArray?
+        get() = getValue(KEY_L1KEY)
+        set(value) {
+            updateValue(KEY_L1KEY, value)
+        }
+
+    override var encryptedL2Key: ByteArray?
+        get() = getValue(KEY_ENCRYPTED_L2KEY)
+        set(value) {
+            updateValue(KEY_ENCRYPTED_L2KEY, value)
+        }
+
+    private fun updateValue(
+        key: String,
+        value: ByteArray?
+    ) {
+        Base64.encodeToString(value, Base64.DEFAULT).also {
+            encryptedPreferences?.edit(commit = true) {
+                if (value == null) remove(key)
+                else putString(key, it)
+            }
+        }
```

Comment:
I would reverse the order (also using okio)
```suggestion
        encryptedPreferences?.edit(commit = true) {
            if (value == null) remove(key)
            else putString(key, value.toByteString().base64())
        }
```